### PR TITLE
Minor optimization for type analysis

### PR DESCRIPTION
**Pull request by GustavoLCR**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/radareorg/radare2/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/radareorg/radare2/blob/master/DEVELOPERS.md#code-style)
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [radare2 book](https://github.com/radareorg/radare2book) with the relevant information (if needed)

**Detailed description**
* Don't pollute debug trace after `aft`
* Reserve a big enough hashtable to avoid rehashing
* Fix `esil->db_trace` not being initialized

Before:
`r2 -ebin.dbginfo=false -Qcaaa libr/asm/libr_asm.so  40.52s user 0.34s system 99% cpu 40.872 total`

After:
`r2 -ebin.dbginfo=false -Qcaaa libr/asm/libr_asm.so  36.98s user 0.34s system 99% cpu 37.339 total`

### CODE REVIEW

**Comment by XVilka** [comment]
Please rebase.

File: test/db/anal/x86_64 [hunk 1]
```diff
@@ -3208,19 +3208,19 @@ c 0x5f8 case.0x59d.9
...
-0x000005b2 0x000005c0 01:0099 14 j 0x000005ab
-0x000005c0 0x000005ce 01:009A 14 j 0x000005ab
-0x000005ce 0x000005dc 01:009B 14 j 0x000005ab
-0x000005dc 0x000005ea 01:009C 14 j 0x000005ab
-0x000005ea 0x000005f8 01:009D 14 j 0x000005ab
-0x000005f8 0x00000606 01:009E 14 j 0x000005ab
-0x00000606 0x00000614 01:009F 14 j 0x000005ab
-0x00000614 0x00000622 01:00A0 14 j 0x000005ab
-0x00000622 0x00000633 01:00A1 17 j 0x000005ab
+0x00000580 0x0000058d 00:0000 13 j 0x00000622 f 0x0000058d
```
Comment (trufae) [inline test/db/anal/x86_64:3211]: looks wrong

File: libr/core/anal_tp.c [hunk 2]
```diff
@@ -492,22 +493,28 @@ R_API void r_core_anal_type_match(RCore *core, RAnalFunction *fcn) {
...
-	RAnalEsilTrace *trace = r_anal_esil_trace_new (core->anal->esil);
-	if (!trace) {
-		return;
-		r_anal_emul_restore (core, hc);
-	}
-	core->anal->esil->trace = trace;
+
+	// Reserve bigger ht to avoid rehashing
+	HtPPOptions opt = core->anal->esil->trace->db->ht->opt;
+	ht_pp_free (core->anal->esil->trace->db->ht);
```
Comment (trufae) [inline libr/core/anal_tp.c:510]: Hold the trace pointer once and avoid so many repeated dereferences

**Review by trufae — CHANGES_REQUESTED** [review]

File: test/db/anal/x86_64 (hunk 1 above)
Comment (GustavoLCR) [reply test/db/anal/x86_64:3238]: This is 489215f, I think this should be the correct behavior

**Review by XVilka — APPROVED** [review]

**Comment by ret2libc** [comment]
@trufae please review again.

File: test/db/anal/x86_64 (hunk 1 above)
Comment (trufae) [reply test/db/anal/x86_64:3238]: Do we have any test that is actually testing that the tracing is not broken? im mainly worried about the change in behaviour here. its fine to have zero es around if this tracing info is generated by an automatic analysis command. if theres no such test. can you add it? 

Thanks!!

**Review by trufae — CHANGES_REQUESTED** [review]
Also add ## and performance details (is this minor optimization noticeable?)

**Review by trufae — APPROVED** [review]

**Comment by trufae** [comment]
LGTM!

**Pull request merged**